python/lsst/obs/base/cli/cmd/register.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of obs_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import logging
+
+from lsst.daf.butler.cli.opt.repo import repo_option
+from lsst.obs.base.cli.opt import instrument_option
+from lsst.daf.butler import Butler
+from lsst.utils import doImport
+
+__all__ = ['register']
+
+
+@click.command()
+@instrument_option(required=True, helpMsg="The fully-qualified name of an Instrument subclass.")
+@repo_option(required=True)
+@click.pass_context
+def register(context, repo, instrument):
+    """Add an instrument to the data repository.
+    """
+    try:
+        instr = doImport(instrument)
+    except Exception as err:
+        if '.' not in instrument:
+            fullQualMsg = "'register' requires a fully-qualified instrument name. "
+        else:
+            fullQualMsg = ""
+        logging.critical("Could not import instrument: %s. %sFailed with exception: %s",
+                         instrument, fullQualMsg, err)
+        context.exit(1)
+    try:
+        instr = instr()
```

Comment:
This should never fail if we have an `Instrument`. What might be worth doing is checking that what you have is actually an instance of `Instrument` since you are letting people pass in random classes.  This code should be refactored though into a standalone routine that will also deal with the `Instrument.fromName` scheme. If the supplied registry is `None` you can just do what you do here.